### PR TITLE
[CIVIS-1822] ENH `file_id_from_run_output` works for all job types

### PR DESCRIPTION
**Pull request by KasiaHinkson**
Replace `client.scripts.list_containers_runs_outputs` with `client.jobs.list_runs_outputs` in `civis.io.file_id_from_run_output` so it works for all job types, not just container scripts. 

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

**Comment by KasiaHinkson** [comment]
Done, that worked. Thank you!

File: civis/parallel.py [hunk 1]
```diff
@@ -710,7 +710,7 @@ def _fetch_result(fut):
 
             try:
                 # Find the output file ID from the run outputs.
-                run_outputs = client.scripts.list_containers_runs_outputs(
+                run_outputs = client.jobs.list_runs_outputs(
```
Comment (jacksonlee-civis) [inline civis/parallel.py:713]: This `_CivisBackendResult` class is designed to work with only _container_ scripts, so I don't think we should modify `_CivisBackendResult`'s implementation.

If you take back this change and have both `scripts.list_containers_runs_outputs` and `jobs.list_runs_outputs` mocked out at the `create_client_mock_for_container_tests` function below, then the test suite will pass.

File: civis/tests/mocks.py [hunk 2]
```diff
@@ -75,7 +75,7 @@ def create_client_mock_for_container_tests(
         mock_container_run['error'] = 'None'
     c.scripts.post_containers_runs.return_value = mock_container_run_start
     c.scripts.get_containers_runs.return_value = mock_container_run
-    c.scripts.list_containers_runs_outputs.return_value = (run_outputs or [])
+    c.jobs.list_runs_outputs.return_value = (run_outputs or [])
```
Comment (jacksonlee-civis) [inline civis/tests/mocks.py:78]: Related to the comment above -- try mocking both `scripts.list_containers_runs_outputs` and `jobs.list_runs_outputs`.

**Review by jacksonlee-civis — COMMENTED** [review]
@KasiaHinkson Thank you for improving civis-python! Two small changes are needed (see my comments below), then this PR should be good to go.

File: civis/tests/mocks.py (hunk 2 above)
Comment (KasiaHinkson) [reply civis/tests/mocks.py]: That makes complete sense, thank you!

**Review by jacksonlee-civis — APPROVED** [review]
LGTM!

**Pull request merged**